### PR TITLE
BUG: Add ability to override non-empty list params with empty

### PR DESCRIPTION
**Pull request by mebristo**
Fixes [the following issue](https://github.com/microsoft/hi-ml/issues/329) i.e. allows us to override list parameters that are not empty by default with empty lists

### CODE REVIEW

File: hi-ml-azure/testazure/testazure/test_azure_util.py [hunk 1]
```diff
@@ -1891,6 +1893,27 @@ class SimpleClass(param.Parameterized):
...
+    (["--strings='None'"], "strings", ["'None'"], True),
+    (["--strings=','"], "strings", ["'", "'"], True),
+    (["--strings=''"], "strings", ["''"], True),
+    (["--strings=,"], "strings", [], True),
+    (["--strings="], "strings", [], True),
+    (["--integers="], "integers", [], True),
+    (["--floats="], "floats", [], True)])
+def test_override_list(parameterized_config_and_parser: Tuple[ParamClass, ArgumentParser],
+                       args: List[str], expected_key: str, expected_value: Any, expected_pass: bool) -> None:
+    """Test different options of overriding a non-empty list parameter to get an empty list"""
```
Comment (ant0nsc) [inline hi-ml-azure/testazure/testazure/test_azure_util.py:1910]: nitpick: All your expected_pass settings are True, why have that parameter?

**Review by ant0nsc — APPROVED** [review]

File: hi-ml-azure/testazure/testazure/test_azure_util.py (hunk 1 above)
Comment (peterhessey) [reply hi-ml-azure/testazure/testazure/test_azure_util.py:1910]: Seconding this: either remove the `expected_pass` variable or add expected failing tests (however for this specific test I don't think that's necessary as the failing tests seem to be covered by `test_create_parser()`

**Review by peterhessey — CHANGES_REQUESTED** [review]
All lgtm, happy to approve once the `expected_pass` variable has been updated

**Comment by mebristo** [comment]
> All lgtm, happy to approve once the `expected_pass` variable has been updated/removed

File: hi-ml-azure/testazure/testazure/test_azure_util.py (hunk 1 above)
Comment (mebristo) [reply hi-ml-azure/testazure/testazure/test_azure_util.py:1910]: Good point, removed

**Comment by mebristo** [comment]
Reopen

**Comment by mebristo** [comment]
Reopen

**Comment by mebristo** [comment]
>

**Comment by mebristo** [comment]
(Investigating why PR gets closed)

**Review by peterhessey — APPROVED** [review]

**Pull request merged**